include/clang/AST/ASTContext.h

```diff
@@ -2333,6 +2320,35 @@ class ASTContext : public RefCountedBase<ASTContext> {
   void ResetObjCLayout(const ObjCContainerDecl *CD);
 
   //===--------------------------------------------------------------------===//
+  //                    Predicates For Checked C checked types
+  //===--------------------------------------------------------------------===//
+
+  /// \brief Determine whether this pointer, array or function type
+  /// provides at least as much checking as the other type.  Return true if
+  /// it does or false if it does not or the types differ in some other way
+  /// than checkedness.
+  bool isAtLeastAsCheckedAs(QualType T1, QualType T2) const;
```

Comment:
It's great to see the function name implying the direction of the comparison between `T1` and `T2`, but I might suggest explicitly adding to the comment that you're checking that `T1` is at least as checked as `T2`.